mempool/mempool
# If block indexing fails | chain of hash invalid, try again in 10 seconds

Fixes https://github.com/mempool/mempool/issues/1915

### Testing

* Run bitcoin core
* Make sure you have many blocks to index (at least 1,000 on mainnet, 10,000 on testnet/signet)
* Run the node backend
* Wait for the block indexing to start
```
Jun 25 19:31:40 [49459] DEBUG: Indexing blocks from #2282488 to #0
Jun 25 19:31:41 [49459] INFO: Indexing 9566 blocks from #2022488 to #2012489
Jun 25 19:31:45 [49459] DEBUG: Indexing block #2021608 | ~74.50 blocks/sec | total: 260881/2282489 (11.43%) | elapsed: 6 seconds | left: ~27136 seconds
```
* Suspend or stop bitcoin core
* Wait for the the node backend connection to core to timeout or fail
```sh
### If you suspend you'll see this
Jun 25 19:33:25 [49459] ERR: Block indexing failed. Trying again in 10 seconds. Reason: ETIMEDOUT
Jun 25 19:33:27 [49459] DEBUG: updateMempoolInfo: ETIMEDOUT
### If you stop you'll see this (it's faster to test this way)
Jun 25 19:33:25 [49459] ERR: Block indexing failed. Trying again in 10 seconds. Reason: connect ECONNREFUSED 127.0.0.1:8332
Jun 25 19:33:27 [49459] DEBUG: updateMempoolInfo: connect ECONNREFUSED 127.0.0.1:8332
```
* Resume bitcoin core
* Confirm the block indexing restarts
```
Jun 25 19:34:04 [49459] DEBUG: Indexing blocks from #2282488 to #0
Jun 25 19:34:06 [49459] INFO: Indexing 5611 blocks from #2022488 to #2012489
Jun 25 19:34:10 [49459] DEBUG: Indexing block #2017688 | ~68.67 blocks/sec | total: 264801/2282489 (11.6%) | elapsed: 6 seconds | left: ~29384 seconds
```
